### PR TITLE
Improvements on Unit Tests for Community, fixed bugs in GetCommunitiesByPublicityViewController and See Post functionality

### PR DESCRIPTION
**Pull request by irem-zeynep**
- Naming and Javadoc improvements
- Configured the mock service for the _exists_ method in **CreateCommunityControllerTest**
-  Added test for returning 403 when community name is not unique in  **CreateCommunityControllerTest**
- Verified correct methods are called with correct parameters exactly once and no other methods are called during the tests.
- Fixed sending parameter with the redirect in **GetCommunitiesByPublicityViewController**
- Fixed bug in see post functionality: **GetPostsViewController**. and **posts** template _(Fixed casting problem in the controller and data retrieval problem in the template)_

### CODE REVIEW

**Review by Ufuk-Arslan — APPROVED** [review]
Here are my thoughts on your pull request:
- Renaming the methods and giving them proper parameters will surely help us understand their purpose better and eliminate confusion. 
- Making sure that tests are using just the required methods will remove inconsistencies.
- Checking for specific errors and returning the expected status code is essential when spotting the problem when we face one. So checking whether the community name is unique or not will be helpful.

Overall your code is very well implemented. Thanks for the enhancements and your efforts.

**Review by schroscatt — COMMENTED** [review]
My suggestion:
You don't have to include createCommunity_ or getCommunitiesByPublicity_ in the beginning of the function names because they are the same as the name of the Class they belong to. Also too long names make the code less readable. 
If you have other reasons to give those names I'd appreciate if you can share with me

**Review by schroscatt — APPROVED** [review]

**Comment by irem-zeynep** [comment]
> My suggestion:
> You don't have to include createCommunity_ or getCommunitiesByPublicity_ in the beginning of the function names because they are the same as the Class name they belong to. Also too long names make the code less readable.
> If you have other reasons to give those names I'd appreciate if you can share with me

Thanks for the feedback!

I followed this [blog](https://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html). 
I agree with you about long names. Hence, instead of writing expected behavior in name with detail, I commented on the test. I included the names considering the new method additions. Also, I wanted to follow the same naming convention as communityServiceTests :) 
But if you suggest me to change, I can. It would be nicer if we all use the same convention.

**Review by orhungorkem — APPROVED** [review]
Redirecting publicity was necessary to keep the option set by user while searching by publicity. This seems fine, thank you.

File: practiceapp/src/main/java/com/practiceapp/practiceapp/controller/views/GetPostsViewController.java [hunk 1]
```diff
@@ -27,7 +28,12 @@
     @RequestMapping(path = "/{community_name}",method = RequestMethod.GET)
     public String showPostsByCommunityName(Model model, @PathVariable("community_name") String community_name){  //TEST AFTER POSTS ADDED!
 
-        List<PostEntity> posts = restTemplate.getForObject(base_url + target_url + "/" + community_name, List.class);
+        System.out.println("NEREDE");
+        System.out.println("NEREDE");
+        System.out.println("NEREDE");
```
Comment (orhungorkem) [inline practiceapp/src/main/java/com/practiceapp/practiceapp/controller/views/GetPostsViewController.java:33]:
```suggestion
       
```

File: practiceapp/src/main/java/com/practiceapp/practiceapp/controller/views/GetPostsViewController.java [hunk 2]
```diff
@@ -38,7 +44,9 @@ public String showPostsByCommunityName(Model model, @PathVariable("community_nam
 
             rows.put(row);
         }
-        model.addAttribute("posts", rows);
+
+        System.out.println(rows);
```
Comment (orhungorkem) [inline practiceapp/src/main/java/com/practiceapp/practiceapp/controller/views/GetPostsViewController.java:48]:
```suggestion
        
```

**Review by orhungorkem — APPROVED** [review]
I suggested to delete debug prints. On the other hand, thanks for corrections, I suppose there were problems with data types before.

File: practiceapp/src/main/java/com/practiceapp/practiceapp/controller/views/GetPostsViewController.java (hunk 1 above)
Comment (irem-zeynep) [reply practiceapp/src/main/java/com/practiceapp/practiceapp/controller/views/GetPostsViewController.java:33]: I forgot to delete :D Thanks for noticing! I'm deleting it now.

**Pull request merged**